default/python/AI/FreeOrionAI.py

```diff
@@ -202,6 +226,16 @@ def generateOrders():  # pylint: disable=invalid-name
     at end of this function, fo.doneTurn() should be called to indicate to the client that orders are finished
     and can be sent to the server for processing."""
     empire = fo.getEmpire()
+    if empire is None:
+        print "This client has no empire. Doing nothing to generate orders."
+        try:
+            # early abort if no empire. no need to do meter calculations
+            # on last-seen gamestate if nothing can be ordered anyway...
+            fo.doneTurn()
+        except Exception as e:
```

Comment:
This looks odd, but is correct.  I'd add a comment that `doneTurn()` is issued on behalf of the client network id, not the empire id, so not having a correct empire id does not invalidate `doneTurn()`